sysid/modelo.m

```diff
@@ -0,0 +1,279 @@
+    aux(i,:)=orbx_val_div{1,i};
+    orbx_val_avg=mean(aux,1)';
+end
+sublengths = ones(1, floor(numel(orby_val)/n))*n;
+orby_val_div = mat2cell(orby_val', 1, sublengths);
+for i=1:numel(sublengths)
+    aux(i,:)=orby_val_div{1,i};
+    orby_val_avg=mean(aux,1)';
+end
+sublengths = ones(1, floor(numel(currdata_val)/n))*n;
+currdata_val_div = mat2cell(currdata_val', 1, sublengths);
+for i=1:numel(sublengths)
+    aux(i,:)=currdata_val_div{1,i};
+    currdata_val_avg=mean(aux,1)';
+end
+%}
+
+%Implementação com sigcond (de repositório fofb/sysid)
+%{
+b = (2^double(prbs_n)-1)*double(prbs_duration); %n blocks -> siglen (prbs), N*duração do estado
+currdata_avg = sigcond(currdata, b, iir_lowpass, fir_remove_switching, [0,0]);
+orbx_avg = sigcond(orbx, b, iir_lowpass, fir_remove_switching, [0,0]);
+%}
+
+%Plot dos sinais avg
+%{
+figure;
+plot(linspace(0,length(currdata_avg)*sampling_frequency*b,length(currdata_avg)),currdata_avg)
+title('currdata avg')
+figure;
+hold on;
+plot(linspace(0,length(orbx_avg)*sampling_frequency*b,length(orbx_avg)),orbx_avg)
+plot(linspace(0,length(orby_avg)*sampling_frequency*b,length(orby_avg)),orby_avg)
+title('orbx e orby avg')
+%}
+
+%Plots de visualização
+%{
+%FFT dos sinais
+Fs=sampling_frequency;
+L=length(currdata);
+Y1=fft(currdata);
+figure;
+hold on;
+plot(Fs/L*(0:L-1),abs(Y1))
+L=length(currdata_avg);
+Y2=fft(currdata_avg);
+plot(Fs/L*(0:L-1),abs(Y2))
+title('FFT currdata')
+
+Fs=sampling_frequency;
+L=length(orbx);
+Y1=fft(orbx);
+figure;
+hold on;
+plot(Fs/L*(0:L-1),abs(Y1))
+L=length(orbx_avg);
+Y2=fft(orbx_avg);
+plot(Fs/L*(0:L-1),abs(Y2))
+title('FFT orbx')
+
+Fs=sampling_frequency;
+L=length(orby);
+Y1=fft(orby);
+figure;
+hold on;
+plot(Fs/L*(0:L-1),abs(Y1))
+L=length(orby_avg);
+Y2=fft(orby_avg);
+plot(Fs/L*(0:L-1),abs(Y2))
+title('FFT orby')
+%}%{
+%Periodograma currdata
+Fs=sampling_frequency;
+L=length(currdata);
+[Pxx,F] = periodogram(currdata,hamming(length(currdata)),length(currdata),Fs);
+figure;
+hold on;
+plot(F,10*log10(Pxx))
+
+Fs=sampling_frequency;
+L=length(currdata_avg);
+[Pxx,F] = periodogram(currdata_avg,hamming(length(currdata_avg)),length(currdata_avg),Fs);
+%figure;
+plot(F,10*log10(Pxx))
+title('Periodograma currdata')
+
+%Periodograma orbx
+Fs=sampling_frequency;
+L=length(orbx);
+[Pxx,F] = periodogram(orbx,hamming(length(orbx)),length(orbx),Fs);
+figure;
+hold on;
+plot(F,10*log10(Pxx))
+
+Fs=sampling_frequency;
+L=length(orbx_avg);
+[Pxx,F] = periodogram(orbx_avg,hamming(length(orbx_avg)),length(orbx_avg),Fs);
+%figure;
+plot(F,10*log10(Pxx))
+title('Periodograma orbx')
+
+%Periodograma orby
+Fs=sampling_frequency;
+L=length(orby_H);
+[Pxx,F] = periodogram(orby,hamming(length(orby)),length(orby),Fs);
+figure;
+hold on;
+plot(F,10*log10(Pxx))
+
+Fs=sampling_frequency;
+L=length(orby_H_avg);
+[Pxx,F] = periodogram(orby_avg,hamming(length(orby_avg)),length(orby_avg),Fs);
+%figure;
+plot(F,10*log10(Pxx))
+title('Periodograma orby')
+%}
+
+%Aplicação no modelo ARX
+if plano=='X'
```

Comment:
Since only one of the orbit planes are being used, what if you pick the appropriate one at the beginning of the function and only apply the filtering/averaging to it? This way you don't loose processing time with a signal that you won't use.